masterplan_tools/method/publicspace/ps_greedy.py

```diff
@@ -0,0 +1,612 @@
+
+        M = ["Клуб для детей и подростков", "Торгово-развлекательный центр", "Парк развлечений", "Зоопарк"]
+        U = ["Музыкальная школа", "Библиотека", "Музей"]
+        S = [
+            "Театр/Концертный зал",
+            "Музей",
+            "Кинотеатр",
+            "Квест",
+            "Картинная галерея",
+            "Арт пространство",
+            "Аквапарк",
+            "Цирк",
+            "Боулинг",
+        ]
+        Sp = ["Спортивная секция", "Спортивный центр", "Бассейн", "Стадион"]
+
+        type_dic = {}
+        for i in a.id_left.unique():
+            type_dic[int(i)] = {"Многофункциональное": 0, "Событийное": 0, "Спортивное": 0, "Учебное": 0}
+        for id, row in a.iterrows():
+            key = row["id_left"]
+            zone_name = row["city_service_type"]
+            if zone_name in M:
+                type_dic[key]["Многофункциональное"] += 1
+            if zone_name in U:
+                type_dic[key]["Учебное"] += 1
+            if zone_name in S:
+                type_dic[key]["Событийное"] += 1
+            if zone_name in Sp:
+                type_dic[key]["Спортивное"] += 1
+
+        new_dic = {}
+        for key, value in type_dic.items():
+            max = 0
+            max_type = "Многофункциональное"
+            for type, v in value.items():
+                if v > max:
+                    max = v
+                    max_type = type
+            new_dic[key] = max_type
+
+        for key, value in new_dic.items():
+            index = self.OPM[self.OPM["id"] == key].index[0]
+            self.OPM.loc[index, "functional_type"] = value
+            self.OPM.loc[index, "exists"] = "существующее"
+
+    # Функция строит слой с изхронами для указанного времени
+    def get_isohrone_time(self, time):
+        street_graph = self.street_graph
+        # buildings = gpd.read_file(objects)
+        zones = self.OPM
+
+        def nodes_finder(street_graph, objects):
+            nodes = ox.distance.nearest_nodes(street_graph, objects.geometry.x, objects.geometry.y)
+            return nodes
+
+        def isochrones_generator_time(street_graph, obj_node, DIST):
+            subgraph = nx.ego_graph(street_graph, obj_node, radius=DIST, distance="time")
+            node_points = [shapely.geometry.Point((data["x"], data["y"])) for node, data in subgraph.nodes(data=True)]
+            isochrones = gpd.GeoSeries(node_points).unary_union.convex_hull
+            return isochrones
+
+        zones_points = zones.copy()
+        zones_points = zones_points.to_crs(32636)
+        zones_points["geometry"] = zones_points["geometry"].centroid
+        zones_points["node"] = nodes_finder(street_graph, zones_points)
+
+        zones_iso_time = zones_points
+        zones_iso_time.geometry = zones_iso_time.node.apply(lambda x: isochrones_generator_time(street_graph, x, time))
+        """
+        Необходимо построить изохроны для разных временных радиусов - 5, 10,  20,  минут.
+        """

```

Comment:
And this on contrary should not be an attempt to create a dicsting, but a simple code comment with #